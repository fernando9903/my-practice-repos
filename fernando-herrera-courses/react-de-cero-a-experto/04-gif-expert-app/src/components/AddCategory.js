import React, { useState } from "react";

export const AddCategory = () => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const handleOnSubmit = (event) => {
        event.preventDefault();
        console.log('Submit hecho');
        setInputValue('');
    }

    return (
        <form onSubmit={handleOnSubmit}>
            <input
                type='text'
                placeholder='Type a new category that you want to add and press enter'
                onChange={handleInputChange}
            />
        </form>
    );
}