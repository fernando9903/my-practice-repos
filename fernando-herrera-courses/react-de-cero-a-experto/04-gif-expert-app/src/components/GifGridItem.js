import React from "react";
import PropTypes from 'prop-types';

export const GridItem = ({title, url}) => {

    return (
        <div className='card'>
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}