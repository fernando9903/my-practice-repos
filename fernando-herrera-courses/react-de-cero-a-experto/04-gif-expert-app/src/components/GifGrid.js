import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";

import { GridItem } from "./GifGridItem";

export const GifGrid = ({category}) => {

    const {loading, data:images} = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {loading ? 'Loading...' :
            <div className='card-grid animate__animated animate__fadeInTopLeft'>
                {images.map( image => 
                    <GridItem
                        key={image.id}
                        {...image}
                    />
                )}
            </div>}
        </>
    )
}