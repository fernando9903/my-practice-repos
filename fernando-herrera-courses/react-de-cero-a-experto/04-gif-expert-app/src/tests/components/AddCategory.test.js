import { shallow } from "enzyme";
import { AddCategory } from "./../../components/AddCategory";

describe('Tests on the AddCategory component', () => {
    
    const setCategory = () => {}
    const wraper = shallow(<AddCategory setCategories = {setCategory}/>);

    test('The componet renders correctly', () => {    
        expect(wraper).toMatchSnapshot();
    });

    test('The text input value change', () => {
        const e = {
            target: {
                value: 'Hello world!'
            }
        }
        const input = wraper.find('input');
        input.simulate('change', e);
    });
})