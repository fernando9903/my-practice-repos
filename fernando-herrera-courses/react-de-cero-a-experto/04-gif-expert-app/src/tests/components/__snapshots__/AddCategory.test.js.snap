// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on the AddCategory component The componet renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Type a new category that you want to add and press enter"
    type="text"
    value=""
  />
</form>
`;
